.github/workflows/pull_request.yml

```diff
@@ -82,3 +82,82 @@ jobs:
         env:
           IMG_RESULT: cache
         run: make docker.buildx.migrate
+
+  unit-tests:
+    name: unit tests 
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Set up Go 1.14
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.14.7
+      id: go
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: verify license
+      run: make check-license 
+
+    - name: verify dependencies
+      run: make deps
+
+    - name: verify tests
+      run: make test
+
+  migration-e2e:
+    needs: ['unit-tests']
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        kubernetes: [v1.18.6]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      
+      - name: Setup Minikube-Kubernetes
+        uses: manusa/actions-setup-minikube@v2.3.0
+        with:
+          minikube version: v1.9.2
+          kubernetes version: ${{ matrix.kubernetes }}
+          github token: ${{ secrets.GITHUB_TOKEN }}
+      
+      - name: Set tag
+        run: |
+          BRANCH="${GITHUB_REF##*/}"
+          CI_TAG="ci"
+          echo "TAG=${CI_TAG}" >> $GITHUB_ENV
+          echo "BRANCH=${BRANCH}" >> $GITHUB_ENV
+
+      - name: Running tests
+        run: ./ci/migrate/sanity.sh
```

Comment:
@shubham14bajpai How will this work? we need to build the image and test that image? Here it is fetching from the registry.

Reply:
IMHO, these should be the steps:
1. Build the multiplatform image (Job1)
2. Push it to some regsitry, either a local one using `registry:2` or ghcr
3. Run minikube (Job2) and pull images from this registry and test it. 

Though the multiarch images will be built, only the amd64 version (i.e the one which runs minikube) will be tested, which is fine atm.

Reply:
@akhilerm the sanity test has a make command which builds a -amd64 image locally. But I will switch to your suggestion  :+1:

Reply:
We can follow either approach, if the steps mentioned are not easily doable. Will first migrate the tests to github actions and then make it run with the same image in all cases.